File: frontend/types/applicationEvent.ts
export type ApplicationEvent = {
  id: number;
  application_id: number;
  event_type: string;
  source: string;
  old_status: string | null;
  new_status: string | null;
  message: string | null;
  gmail_message_id: string | null;
  created_at: string;
  company: string | null;
  position: string | null;
};

File: backend/app/schemas/application_event.py
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class ApplicationEventOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    application_id: int
    event_type: str
    source: str
    old_status: str | None = None
    new_status: str | None = None
    message: str | None = None
    gmail_message_id: str | None = None
    created_at: datetime
    company: str | None = None
    position: str | None = None